lib/restrictions.js

```diff
@@ -57,6 +62,17 @@ export async function insert({restriction} = {}) {
     meta,
     restriction
   };
+  if(restriction.method === 'limitOverDuration') {
+    const durationRegex =
+  /^P(\d+Y)?(\d+M)?(\d+W)?(\d+D)?(T(\d+H)?(\d+M)?(\d+S)?)?$/;
+    const duration = restriction.methodOptions.duration;
```

Comment:
I looked into optional chaining, but eslint does not currently support it, even when disabling it for that line. I have added a check for `methodOptions`.

Reply:
Making the eslintrc.js file looks like this will allow for optional chainging to be used: https://gist.github.com/mattcollier/3cff1c1a51138b25d5c7c24387906db0

Also, lehn just put this in: https://github.com/digitalbazaar/eslint-config-digitalbazaar/issues/48

Reply:
This solves it for eslint, however I am still running into issues using optional chaining with the test suite. I had added the same line to the eslintrc.js file, but I am still seeing `SyntaxError: Invalid or unexpected token`.

Reply:
fixed: https://github.com/digitalbazaar/bedrock-resource-restriction/pull/23